components/tools/OmeroWeb/omeroweb/webgateway/views.py

```diff
@@ -1198,6 +1210,40 @@ def wrap(request, *args, **kwargs):
     return wrap
 
 
+def json_response(f):
+    """
+    Decorator for wrapping response in JsonResponse if needed and
+    error handling
+
+    @param f:       The function to wrap
+    @return:        The wrapped function, which will return json
+    """
+    @wraps(f)
+    def wrap(request, *args, **kwargs):
+        logger.debug('json_response')
+        try:
+            rv = f(request, *args, **kwargs)
+            if isinstance(rv, HttpResponse):
```

Comment:
You already cache nicely omero expections https://github.com/openmicroscopy/openmicroscopy/pull/4708/files#diff-fc3a666f42c5099bec2593d6f4b4cf2dR1229 the same should happen for internal web api as well